### ISSUE TITLE
Force browser to not store cache for logout page

### ISSUE THREAD

**Issue by sebj54** [contao/core-bundle#1702]
**Affected version(s)**

4.4.41 (and superior?)

**Description**  

With a logout cache, `Cache-Control` header is set to `private` by default.

If we refer to the documentation on [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#Cacheability), this does not prevent browser from caching the page.

> Indicates that the response is intended for a single user and must not be stored by a shared cache. A private cache may store the response.

If a user disconnect, reconnect and tries to disconnect again (it happens, trust me!), Chrome caches the page (it seems that Firefox doesn't). So the redirection is made correctly but user is not disconnected.

**How to reproduce**  

 - Connect
 - Disconnect
 - Re-connect
 - Try to disconnect again

You may not be able to reproduce because I don't experienced this behaviour on my staging server. Anyway, this `private` value does not seem like a way to avoid caching for this kind of page.

**How to solve**

I tried to solve this bug by adding headers to rhe `RedirectResponse` constructor in [PageLogout.php](https://github.com/contao/core-bundle/blob/master/src/Resources/contao/pages/PageLogout.php):

```php
return new RedirectResponse($strRedirect, 302, [
    'Cache-Control' => 'no-store, no-cache, must-revalidate, post-check=0, pre-check=0',
]);
```

**Comment by aschempp** [comment]
Maybe we should use a 303 redirect instead?

**Comment by sebj54** [comment]
If I'm right, that's what the deprecated `ModuleLogout` does. I had the same problem with this module instead of a `PageLogout`. I switched because of the deprecation and hoped this will fix my issue 😁

**Comment by leofeyer** [comment]
@Toflar /cc

**Comment by Toflar** [comment]
> Anyway, this `private` value does not seem like a way to avoid caching for this kind of page.

That's correct. `private` is not supposed to disable caching.
I don't understand why you think it should?

Yes, if we want to make sure the logout redirect is not cached, then we have to add those headers. However, `post-check` and `pre-check` are nonsense.

**Comment by sebj54** [comment]
You're right, I didn't express myself clearly. Here is what I wanted to say: I think this page should never be cached and I know the `private` value does not avoid cache. 

Now, the values I added in my temporary may not be the correct ones as I rarely play with cache options. To be honest, I found these values on Stack Overflow post.
